### PR TITLE
Respect `add_prefix_space` option in `LlamaTokenizerFast`

### LINKED ISSUE #29625: `add_prefix_space` won't be respected by Llama tokenizer
### System Info

- `transformers` version: 4.38.2
- Platform: Linux-6.5.0-14-generic-x86_64-with-glibc2.35
- Python version: 3.10.13
- Huggingface_hub version: 0.21.3
- Safetensors version: 0.4.2
- Accelerate version: 0.27.2
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

With `sentencepiece==0.2.0` and `protobuf==4.25.3` installed

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
>>> from transformers import AutoTokenizer
>>> tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-hf", local_files_only=True, add_prefix_space=False)
>>> tokenizer.tokenize("overheard")
['▁over', 'he', 'ard']
```

Also tried `add_dummy_prefix_space=False`, the output is still the same.

### Expected behavior

The tokenize result should not add prefix space (`SPIECE_UNDERLINE`)

### PR DESCRIPTION
**Pull request by aoxolotl**
# What does this PR do?
Respect `add_prefix_space` option in Llama tokenizer (Fixes #29625)

The `add_prefix_space` option in Llama tokenizer was set but not passed to the `super().__init__()` method to `PretrainedTokenizersFast`. This resulted in SPIECE_UNDERLINE token being added even when `add_prefix_space=False`. 

## Minimal example
When `add_prefix_space` is `False`
```python
>>> from transformers import AutoTokenizer
>>> tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-hf", add_prefix_space=False)
>>> tokenizer.tokenize('overheard')
['over', 'he', 'ard']
```

`add_prefix_space` is `True` 
```python
>>> tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-hf", add_prefix_space=True)
>>> tokenizer.tokenize('overheard')
>>> ['\u2581over', 'he', 'ard']
```

Fixes #29625 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@ArthurZucker @scruel

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Could you add a test to make sure this works? 🤗

**Comment by aoxolotl** [comment]
Sure! Will update the pull request with the tests

**Comment by aoxolotl** [comment]
Hey @ArthurZucker, I have a question about the tests. There already seems to be an `add_prefix_space` test present in `test_tokenization_llama.py`. However, this strangely passes even without the modifications above. The difference seems to come from using `AutoTokenizer` (as in the mentioned issue) vs using the `LlamaTokenizer*` classes directly. Is there a difference between the `init` routes we take in the above two scenarios?

**Comment by ArthurZucker** [comment]
Actually I think we need to wait for #28881, to have a proper fix for Llama! Feel free to skip it. T5 should work!

**Comment by scruel** [comment]
Won't you see my review?
image

**Comment by ArthurZucker** [comment]
pending means it's not been submitted!

**Comment by scruel** [comment]
> pending means it's not been submitted!

Oh, I see, it this HF repos limited review action by others?

**Comment by ArthurZucker** [comment]
No, pressing the `submit review` button should be enough

**Comment by scruel** [comment]
> No, pressing the `submit review` button should be enough

It works! Thanks!

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -737,6 +737,7 @@ def as_tensor(value, dtype=None):
 
             def is_tensor(obj):
                 return isinstance(obj, mx.array)
+
```
[the remark this answers is not in the document]
Comment (aoxolotl) [reply src/transformers/tokenization_utils_base.py]: Sure, updating in latest commit

**Comment by aoxolotl** [comment]
@scruel Removed in latest commit!
@ArthurZucker I will make a separate PR for T5 tokenizer which adds any necessary tests. Leaving this PR for `LlamaTokenizer` specific changes

**Comment by ArthurZucker** [comment]
#28881  should actually fix this!

**Comment by psinger** [comment]
@ArthurZucker are you sure this is solved? I am still experiencing this issue in 4.41.0

**Pull request closed without merging**